File: python/greedy_algorithms/until_number_one.py
# import time
# start_time = time.time()

N, K = map(int, input().split())
result = 0

while True:
    # 나누어 떨어질때까지 1을 뺀다
    count = (N//K) * K
    result += (N - count)

    N = count

    if N < K:
        break

    result += 1
    N //= K

result += (N-1)
print(result)

# end_time = time.time()
# print("time :", end_time - start_time)

File: greedy/books/until_number_one.cpp
#include<iostream>
#include<algorithm>
#include<ctime>
using namespace std;

int main_3()
{
	//time_t start_time = time(0);
	
	int N, K;
	cin >> N >> K;

	int count = 0;

	//N이 K보다 값이 크면 계속 나누기
	while (N >= K)
	{
		if (N % K == 0)
			N /= K;
		else
			N -= 1;
		count++;
	}

	count += N - 1;

	cout << count << endl;
	/*
	time_t end_time = time(0);

	cout << "time : " << end_time - start_time << endl;
	time = 2*/

	return 0;
}